subsys/testsuite/ztest/src/ztest.c

```diff
@@ -427,6 +427,41 @@ void end_report(void)
 K_APPMEM_PARTITION_DEFINE(ztest_mem_partition);
 #endif
 
+struct unit_test_node *z_ztest_unit_test_head;
+
+int ztest_run_registered_test_suites(const void *state)
+{
+	struct unit_test_node *prev_node = NULL;
+	struct unit_test_node *current_node = z_ztest_unit_test_head;
+	int count = 0;
+
+	while (current_node) {
+		bool should_run = true;
+
+		if (current_node->pragma) {
+			should_run = current_node->pragma(state);
+		}
+		if (should_run) {
+			z_ztest_run_test_suite(current_node->name, current_node->suite);
+			++count;
+		}
+		/* Remove the node if pragma is NULL. This ensures that the suite runs only once. */
+		if (!current_node->pragma) {
+			if (prev_node) {
```

Comment:
Violation: Rule 14.4
```suggestion
			if (prev_node != NULL) {
```